__tests__/app.test.js

```diff
@@ -168,7 +168,68 @@ describe("GET /api/articles/:article_id/comments", () => {
   });
 });
 
+describe("POST", () => {
+  test("POST 201: adds a new comment using username and body from request", () => {
+    const testInput = {
+      username: "butter_bridge",
+      body: "really important words about thoughts"
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(testInput)
+      .expect(201)
+      .then(({ body }) => {
+        const { comment } = body;
+        expect(comment).toMatchObject({
+          comment_id: expect.any(Number),
+          body: expect.any(String),
+          author: expect.any(String),
+          article_id: expect.any(Number),
+          votes: expect.any(Number),
+          created_at: expect.any(String)
+        });
+      });
+  });
+  test("POST 400: responds with invalid request error message when post request is sent with an empty object", () => {
+    const testInput = {}
+      return request(app)
+      .post("/api/articles/1/comments")
+      .send(testInput)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("status 400: invalid request, missing information");
+      })
+  })
+  test("POST 400: responds with invalid request error message when post request is sent with an empty property in the object", () => {
+    const testInput = {
+      username: "butter_bridge", 
+    }
+      return request(app)
+      .post("/api/articles/1/comments")
+      .send(testInput)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("status 400: invalid request, missing information");
+      })
+  })
+  test("GET 404: responds with message and error if input article_id does not have an article attached", () => {
+    return request(app)
+      .get("/api/articles/300/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("status 404: Article not found");
+      });
+  });
+  test("GET 404: responds with message and error if there is a typo in the path", () => {
+    return request(app)
+      .get("/api/articles/1/commen")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("status 404: Path not found");
+      });
+  });
+});
```

Comment:
Those are some brilliantly written tests! There are also some more conditions that you should be testing for:
- what if there are extra properties in the post body, i.e. `banana: "true",` we would hope to see that our server can ignore them and post using only the relevant info 
- what if the username provided doesn't exist in the database 😨 i.e. `username: "hacker",`
- what if the article ID is not valid, i.e. - `/api/articles/womble/comments`